sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3406,42 +3406,1131 @@ lsc_format_ret(__ESIMD_NS::simd<T1, N> Vals) {
-/// See description of template and function parameters in @ref
-/// usm_atomic_update0 "atomic update" operation docs.
-template <atomic_op Op, typename Tx, int N, class T = detail::__raw_t<Tx>>
-__ESIMD_API simd<Tx, N> slm_atomic_update(simd<uint32_t, N> offsets,
-                                          simd_mask<N> mask) {
-  detail::check_atomic<Op, T, N, 0>();
-  const auto si = __ESIMD_NS::get_surface_index(detail::LocalAccessorMarker());
-  return __esimd_dword_atomic0<Op, T, N>(mask.data(), si, offsets.data());
+/// @anchor slm_atomic_update0
+/// @brief Atomic update operation performed on SLM.
+/// No-argument variant of the atomic update operation.
+
+/// simd<T, N>
+/// slm_atomic_update(simd<uint32_t, N> byte_offset,
+///                   simd_mask<N> mask = 1);                   /// (slm-au0-1)
+///
+/// simd<T, N>
+/// slm_atomic_update(simd_view<simd<uint32_t, N>, RegionTy> byte_offset,
+///                   simd_mask<N> mask = 1);                   /// (slm-au0-2)
+///
+
+/// The following functions do the same work as slm_atomic_update(). They accept
+/// a local accessor \p lacc and the atomic update is done from SLM associated
+/// with \p lacc plus \p byte_offset applied to it. If \p byte_offset
+/// is omitted, then zero offset is used.
+
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd<uint32_t, N> byte_offset,
+///                          simd_mask<1> pred = 1);
+///                                                             // (lacc-au0-1)
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd_view<simd<uint32_t, N>, RegionTy> byte_offset,
+///                          simd_mask<1> pred = 1);
+///                                                             // (lacc-au0-2)
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd_mask<1> pred = 1);
+///                                                             // (lacc-au0-3)
+
+/// Usage of cache hints or non-standard operation width N requires DG2 or PVC.
+
+/// simd<T, N>
+/// slm_atomic_update(simd<uint32_t, N> byte_offset,
+///                   simd_mask<N> mask = 1);                   /// (slm-au0-1)
+///
+/// Atomically updates \c N memory locations in SLM indicated by
+/// a vector of offsets, and returns a vector of old
+/// values found at the memory locations before update.
+/// @tparam Op The atomic operation - can be \c atomic_op::inc or
+/// \c atomic_op::dec, \c atomic_op::load.
+/// @tparam T The vector element type.
+/// @tparam N The number of memory locations to update.
+/// @param byte_offset The vector of 32-bit offsets.
+/// @param mask Operation mask, only locations with non-zero in the
+///   corresponding mask element are updated.
+/// @return A vector of the old values at the memory locations before the
+///   update.
+///
+template <atomic_op Op, typename T, int N, class Tx = detail::__raw_t<T>>
```

Comment:
Tx is used only inside the function, not in some SFINAE condition.
Can you please move this `Tx=` to the branch where it is actually used, right before the call of __esimd_dword_atomic0

Reply:
Done.